shardingsphere-governance/shardingsphere-governance-core/src/main/java/org/apache/shardingsphere/governance/core/registry/process/subscriber/ProcessRegistrySubscriber.java

```diff
@@ -98,13 +123,21 @@ public void reportExecuteProcessUnit(final ExecuteProcessUnitReportEvent event)
      */
     @Subscribe
     public void reportExecuteProcess(final ExecuteProcessReportEvent event) {
-        String executionPath = ProcessNode.getExecutionPath(event.getExecutionID());
-        YamlExecuteProcessContext yamlExecuteProcessContext = YamlEngine.unmarshal(repository.get(executionPath), YamlExecuteProcessContext.class);
-        for (YamlExecuteProcessUnit unit : yamlExecuteProcessContext.getUnitStatuses()) {
-            if (unit.getStatus() != ExecuteProcessConstants.EXECUTE_STATUS_DONE) {
+        String executionID = event.getExecutionID();
+        synchronized (executionID) {
+            Map<String, Object> dataMap = event.getDataMap();
+            ExecuteProcessReportContext reportContext = (ExecuteProcessReportContext) dataMap.get(ExecuteProcessConstants.EXECUTE_ID.name());
```

Comment:
The value of `dataMap.get(ExecuteProcessConstants.EXECUTE_ID.name())` is `ExecuteProcessReportContext`? Sorry, I can not catch your intend here. Is that a only way? or Is it necessary to do so?

Reply:
Yes, the value is `ExecuteProcessReportContext`.

Since execution process report might be delayed or ignored for better performance, the related data is cached in `ExecuteProcessReportContext`, it will be used in the whole lifecyle of current sql execution.

I've considered to add a new key-value pair in `dataMap`, considering that there's `EXECUTE_ID` already, so I reused it.